superset-frontend/src/components/DatabaseSelector/DatabaseSelector.test.tsx

```diff
@@ -224,6 +224,18 @@ test('Should database select display options', async () => {
   expect(await screen.findByText('test-mysql')).toBeInTheDocument();
 });
 
+test('should show empty state if there are no options', async () => {
+  const props = createProps();
+  // @ts-ignore
+  props.db = null;
+  render(<DatabaseSelector {...props} />, { useRedux: true });
+  const select = screen.getByRole('combobox', {
+    name: 'Select database or type database name',
+  });
+  userEvent.click(select);
+  expect(screen.queryByText('test-mysql')).not.toBeInTheDocument();
```

Comment:
Right now we are only checking for the absence of the test-mysql text but there is no check about the empty text content.